design/survival/design_survival.qmd

```diff
@@ -0,0 +1,1967 @@
+1. In the covariate adjusted hazard ratio estimate score function, should the regression coefficients really be fixed given the original estimate? I expected them to depend on the parameter $\vartheta$ as well, but both in the paper as well as in the supplementary material they are fixed at $\widehat{\beta}_{\text{L}}$ and $\widehat{\gamma}_{\text{SL}}$ respectively.
+    - Has been clarified, basically fixing the regression coefficients is a more approximate version, which is computationally simpler. We can maybe provide both options e.g. via some expert `control` argument. It would be easier though to just provide the full version.
+    - After discussing in the team, we will just provide the approximate version, because that is the only theoretically justified version in the paper.
+1. Should we make the `p_trt` an argument of `robin_surv()`? 
+    - Currently we just always use the observed proportion of the relevant experimental treatment arm in the data set for the given pairwise comparison.
+1. Should we also show the test statistic and p-value from the hazard ratio estimate?
+    - Currently we just show the logrank test statistic and p-value, and only give the log hazard ratio estimate and standard error
+1. In `RobinCar` the `adjust.CovHR()` method is not consistent with `adjust.LogRank()` regarding the handling of ties: In [line 85-87](https://github.com/mbannick/RobinCar/blob/main/R/adjust-covhr.R#L85) we don't have the correction factor that we have in the log-rank test. We should probably use the correction factor for consistency?
+    - The only downside is that then we are no longer giving the same result as `survival::coxph()`.
+    - It seems this inconsistency is already accepted between `survival::coxph()` and `survival::survdiff()`, so we can just follow the `RobinCar` implementation here.
+
+# Prototype
+
+Here we will now include a more package-ready prototype of the survival analysis implementation, which avoids duplicate code and implements also the above described user interface. Please see the initial design doc [here](survival_initial_code.qmd) for all the details as a starting point when reading.
+
+## Example data
+
+As example data, also here we are going to use the NCCTG Lung Cancer Data set, which is available in the `survival` package.
+We are going to use the male patients as treatment arm $j = 0$ and the female patients as treatment arm $j = 1$.
+
+```{r}
+devtools::load_all()
+library(dplyr)
+library(checkmate)
+dat <- survival::lung |>
+  mutate(
+    status = factor(ifelse(status == 1, "Alive", "Dead")),
+    sex = factor(ifelse(sex == 1, "Male", "Female")),
+    status_numeric = as.numeric(status == "Dead"),
+    strata = factor(ph.ecog)
+  )
+head(dat)
+```
+
+## Extracting variables
+
+This is currently duplicating some code from `predict_counterfactual.lm()`, and we can refactor this later to avoid duplication.
+
+```{r}
+h_prepare_vars <- function(formula, data, treatment) {
```

Comment:
may I ask what is the difference between these two qmd files? they both seem to have the implementations. or we should look at the other PR that would put these functions together instead of looking at this document?

Reply:
just look at this document (this is explained in description)